### DISCUSSION TITLE
Подскажите как добавить сайт в zapret vpn?

### DISCUSSION THREAD

**Discussion by Samoxxxa** [bol-van/zapret#955] [Q&A]
Подскажите пожалуйста можно ли добавить сайт animego.org?
Установлена вот такая версия для windows
Сразу говорю, почитал чуть форум, ничего вообще не понимаю, если найдётся гений который просто скажет куда что написать, буду невероятно благодарен до конца жизни
![Снимок экрана 2024-12-18 045905](https://github.com/user-attachments/assets/10de9bc8-0276-42b3-bba7-82cebeca0fd0)

**Comment by bol-van** [comment]
zapret - это не VPN, и как я везде пишу это программа не для рядового пользователя\
она сложна в  использовании и требует технических навыков

**Reply by trueCylar** [reply]
Здравствуйте, как добавить сайт в исключения для zapret? Сайт lichess.org не работает при включеном zapret, а очень хочется играть в шахматы под ютуб=))

**Reply by numbereleven-a** [reply]
> Здравствуйте, как добавить сайт в исключения для zapret? Сайт lichess.org не работает при включеном zapret, а очень хочется играть в шахматы под ютуб=))

Если нужно чтобы какие то сайты не попадали под случайный обход то в zapret-hosts-user-exclude.txt.
Способ работает для режимов фильтрации MODE_FILTER=hostlist или MODE_FILTER=autohostlist.
Следует учесть нулевую фазу, если в стретегиях используется дурение: synack, syndata, wsize, wssize то на эту фазу не действуют фильтры по hostlist.

**Reply by LycorisWave** [reply]
> zapret - это не VPN, и как я везде пишу это программа не для рядового пользователя она сложна в использовании и требует технических навыков

Жестоко )

**Comment by nialone** [comment]
> буду невероятно благодарен до конца жизни

Привет. Используй браузер Tor. Сайт открывается как минимум. ))

Очень капризный сайт, не смог подобрать правило для 443 порта через blockcheck, ВОобще.
Есть подозрение на утечку DNS и ничего с этим сделать также не могу.
Через браузер показывает "Russia - Cloudflare, Inc.", не знаю насколько это смертельно..
Прошивка на роутере старая, нет поддержки DoT и DoH, плюс telnet выкидывает из сеанса.
Вчера заметил версию для винды, проверял ей. 

Даже интересно стало, у остальных так же?

**Reply by numbereleven-a** [reply]
> > буду невероятно благодарен до конца жизни
> 
> Привет. Используй браузер Tor. Сайт открывается как минимум. ))
> 
> Очень капризный сайт, не смог подобрать правило для 443 порта через blockcheck, ВОобще. Есть подозрение на утечку DNS и ничего с этим сделать также не могу. Через браузер показывает "Russia - Cloudflare, Inc.", не знаю насколько это смертельно.. Прошивка на роутере старая, нет поддержки DoT и DoH, плюс telnet выкидывает из сеанса. Вчера заметил версию для винды, проверял ей.
> 
> Даже интересно стало, у остальных так же?

через запрет работает

**Reply by nialone** [reply]
> через запрет работает

Стабильно? Что за настройки в NFQWS_OPT=...? DNS провайдер?
Попробую потестить на другом железе с таким конфигом.

**Reply by numbereleven-a** [reply]
> > через запрет работает
> 
> Стабильно? Что за настройки в NFQWS_OPT=...? DNS провайдер? Попробую потестить на другом железе с таким конфигом.

буду дома, гляну.

**Reply by numbereleven-a** [reply]
> > через запрет работает
> 
> Стабильно? Что за настройки в NFQWS_OPT=...? DNS провайдер? Попробую потестить на другом железе с таким конфигом.

мгтс мск, doh Cloudflare (Standard), `--dpi-desync=fake --dpi-desync-fooling=badseq,md5sig`

**Reply by nialone** [reply]
1. Разобрался с DNS, реализовал DoH через dnscrypt-proxy2 в роутере. 
2. Спасибо за ответ, помогло в осмыслении >> dpi тут простейший.. и твой вариант у меня не заработал)
3. КОму будет скучно, в папке zapret-winws создаем батник с таким содержимым (выбираем любой 1 вариант), сохраняем и запускаем: 

пчела `start "zapret: http,https,quic" /min "%~dp0winws.exe" --wf-l3=ipv4 --wf-tcp=443 --hostlist-domains=animego.org --dpi-desync=fake,disorder2 --dpi-desync-fooling=md5sig`

мгтс `start "zapret: http,https,quic" /min "%~dp0winws.exe" --wf-l3=ipv4 --wf-tcp=443 --hostlist-domains=animego.org --dpi-desync=fake --dpi-desync-fooling=badseq,md5sig`

Заходит в animego без проблем ... (пост В АРХИВ)
ps: blockcheck так и не заработал ...

**Comment by alk2git** [comment]
обнаружилось странное. При попытке открыть animego.org выдалась стандартная заглушка от провайдера, что сайт содержит материалы для взрослых и открыт быть не может. При дальнейших попытках в autohostlist.log (который у меня дебаг-лог добавления в autohostlist.txt) появились сообщения:
19.12.2024 13:14:13 : animego.org : profile 1 : client 10.x.x.x:58257 : proto tls : incoming RST
19.12.2024 13:14:13 : animego.org : profile 1 : client 10.x.x.x:58257 : proto tls : fail counter 1/3
19.12.2024 13:14:15 : animego.org : profile 1 : client 10.x.x.x:58260 : proto tls : fail counter reset. website is working.
такого ранее не наблюдалось. 
То же самое с сайтом quora.com:
19.12.2024 11:36:53 : www.quora.com : profile 1 : client 10.x.x.x:56968 : proto tls : incoming RST
19.12.2024 11:36:53 : www.quora.com : profile 1 : client 10.x.x.x56968 : proto tls : fail counter 1/3
19.12.2024 11:36:58 : www.quora.com : profile 1 : client 10.x.x.x:56970 : proto tls : fail counter reset. website is working.
Добавление вручную animego.org и quora.com в autohostlist.txt тут же позволило последний открыть. animego.org выдал ошибку 504
Не исключено, что доступ из РФ каким-то образом прикрыт самими владельцами сайта, тогда как из Финляндии и Франции спокойно открывается.
Меня здесь больше беспокоит неверное принятие решения winws. Почему он начал считать заблокированные сайты работающими? Использую последнюю опубликованную версию zapret для Windows

**Reply by bol-van** [reply]
"website is working" возникает, когда идут входящие данные, не признанные как ситуация, похожая на блокировку
Это может быть "не блокировочный" http код. Всякие 504 - это точно не блокировочный код.
Или идет отсылка данных вне sequence окна TLS ClientHello. Это может означать, что первый обмен пакетами не был прерван DPI, а значит он пошел дальше, и это уже не похоже на блок

Если сайт то работает, то блокируется, то автохостлист в этой ситуации не поможет, тк счетчик будет сбрасываться

**Reply by alk2git** [reply]
Ясно, что то, что раньше работало довольно четко и с третьей попытки добавлялось в autohostlist.txt, теперь может неверно интерпретироваться. Сейчас точно также не отработался вход на ntc.party, но отработался на rezka.biz 
Скорее всего, это следствие каких-то модификаций в работе DPI, но убедиться невозможно. Раньше я ни разу не сталкивался с ситуацией, когда заблокированный сайт считался работающим. 
Еще, пользуясь случаем, хочу сказать, что последние версии blockcheck под Windows стали находить как-то уж слишком много вариантов пробоя, большинство из которых, будучи добавленными в cmd-файл, не работают. Либо работают для одного сайта, но не работают для других. Да-да, я знаю, что никто не гарантирует и т.д., но все же хотелось бы добавить в выводы blockcheck больше логики по подбору оптимальной стратегии. Либо использовать путь goodcheckgogo, где скрипт проверяет сайты по списку.

**Reply by alk2git** [reply]
специально провел сеанс blockcheck с сайтом quora.com
Нашлось несколько рабочих стратегий, в целом то же, что я использую для других. При попытке зайти на сайт браузер выдает сообщение о том, что сайт не имеет валидного сертификата, в autohostlist.log все те же сообщения о том, что сайт работает. 
В журнале blockcheck встречаются и сообщения 
curl: (35) Recv failure: Connection reset by peer
и сообщения
curl: (35) OpenSSL/3.4.0: error:0A0000C6:SSL routines::packet length too long
curl: (35) OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to quora.com:443
и совсем немного ошибок по тайм-ауту
Явное добавление в autohostlist.txt сразу позволяет открыть сайт. Все же интересно, с чем связано такое поведение?

**Reply by bol-van** [reply]
Автохостлист не распознает ситуацию подмены сертификата провайдером
Через curl -kv можно посмотреть какая страница выдается
если там мегафон какой-нибудь и заблокировано ФЗ, то это подмена

**Reply by alk2git** [reply]
это мгтс для бюджетников, сертификат не подменяют. Не знаю, что значит "заблокировано ФЗ" 
Команда curl -kv https://quora.com:
`$ curl -kv https://quora.com
* Host quora.com:443 was resolved.
* IPv6: (none)
* IPv4: 34.204.247.192, 34.199.155.95, 34.199.189.233, 34.202.152.76, 34.202.117.244
*   Trying 34.204.247.192:443...
* ALPN: curl offers h2,http/1.1
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (OUT), TLS alert, record overflow (534):
* OpenSSL/3.4.0: error:0A0000C6:SSL routines::packet length too long
* closing connection #0
curl: (35) OpenSSL/3.4.0: error:0A0000C6:SSL routines::packet length too long`

Кстати, blockcheck при подборе стратегии для tls1.3 ни одной для quora.com не нашел

Для tls1.2 все иначе:
`$ curl --tlsv1.2 --tls-max 1.2 -kv https://quora.com
* Host quora.com:443 was resolved.
* IPv6: (none)
* IPv4: 34.199.155.95, 34.202.152.76, 34.202.117.244, 34.204.247.192, 34.199.189.233
*   Trying 34.199.155.95:443...
* ALPN: curl offers h2,http/1.1
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (OUT), TLS alert, record overflow (534):
* OpenSSL/3.4.0: error:0A0000C6:SSL routines::packet length too long
* closing connection #0
curl: (35) OpenSSL/3.4.0: error:0A0000C6:SSL routines::packet length too long`

**Reply by bol-van** [reply]
Тогда надо соглашаться в броузере на небезопасный сайт и смотреть когда он загрузится какой там сертификат
Кем выдан
ФЗ - федеральный закон

**Reply by alk2git** [reply]
Не получается соглашаться в браузере на небезопасный сайт. При попытке открыть http://quora.com открывается https://quora.com и выдается сообщение на экране (Edge):
Подключение к этому сайту не защищено. С сайта quora.com отправлен недопустимый отклик. ERR_SSL_PROTOCOL_ERROR.
Нажав значок i в адресной строке, получаю информацию, что сайт не имеет сертификата.
В Firefox результат чуть подробнее:
Secure Connection Failed
An error occurred during a connection to quora.com. SSL received a record that exceeded the maximum permissible length.
Error code: SSL_ERROR_RX_RECORD_TOO_LONG

Ясно, что DPI гадит, но неясно, почему в одном случае это так, а для других сайтов иначе. Просто потому, что добавляли блокировки в разное время и по-разному? Дома (Ростелеком) таких чудес не происходит, четко отрабатывается недоступность сайта и включение его в autohostlist.txt с третьей попытки.

**Reply by bol-van** [reply]
Без шарка сложно понять. Но вероятно что-то шлется все-таки. Не ожидаемое, а другое.
И на это клиент отвечает. И это значит что "website works". Хотя он не works.
Алгоритм autohostlist не безупречен и не учитывает все возможные варианты как может DPI нагадить

**Reply by k0zyrev** [reply]
@[alk2git](https://github.com/alk2git) удалось пофиксить проблему? У меня сейчас такое же на libgen-rs. Под впн или тор прекрасно работает, с запретом выдает ошибку ссл.

**Reply by alk2git** [reply]
нет, к сожалению. Все точно также избирательно. Один сайт мгновенно добавляется в autohostlist.txt, другой со второй попытки показывает, что сайт работает, и иначе как прямым добавлением в hostlist.txt преодолеть блокировку не удается.
Wireshark я могу попробовать, но боюсь, утону в разборе. Нет столько времени и достаточно сноровки, чтобы понять, что именно идет не так. Но может быть, соберусь. Пока использую zapret только для тех сайтов, для которых он точно работает.
